File: app/src/main/java/com/example/tfg_urjc_appfirstrun/Entities/Week.java
package com.example.tfg_urjc_appfirstrun.Entities;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.UUID;

@Entity(tableName = "weeks")
public class Week {
    @PrimaryKey
    @NonNull
    public String weekId;
    @ColumnInfo(name = "training_owner_Id")
    public String trainingOwnerId;
    @ColumnInfo(name = "number_week")
    public int numberWeek;

    public Week (String trainingOwnerId, int numberWeek) {
        this.weekId = UUID.randomUUID().toString();
        this.trainingOwnerId = trainingOwnerId;
        this.numberWeek = numberWeek;
    }

    public String getWeekId() {
        return weekId;
    }

}

File: app/src/main/java/com/example/tfg_urjc_appfirstrun/Entities/Week.kt
package com.example.tfg_urjc_appfirstrun.Entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.jetbrains.annotations.NotNull
import java.util.*

@Entity(tableName = "weeks")
class Week(trainingOwnerId: String?, numberWeek: Int) {
    @PrimaryKey
    @NotNull
    var weekId: String

    @ColumnInfo(name = "training_owner_Id")
    var trainingOwnerId: String?

    @ColumnInfo(name = "number_week")
    var numberWeek: Int

    init {
        weekId = UUID.randomUUID().toString()
        this.trainingOwnerId = trainingOwnerId
        this.numberWeek = numberWeek
    }
}